core/src/main/scala/org/typelevel/sbt/BspControlPlugin.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.sbt
+
+import sbt.Keys._
+import sbt._
+import sbtcrossproject.CrossPlugin
+import sbtcrossproject.CrossPlugin.autoImport._
+import sbtcrossproject.Platform
+
+object BspControlPlugin extends AutoPlugin {
+  override def trigger = allRequirements
+  override def requires: Plugins = CrossPlugin
+
+  object autoImport {
+    lazy val tlBspPlatforms: SettingKey[Set[Platform]] =
+      settingKey[Set[Platform]](
+        "set of platforms for which BSP is enabled (default: JVMPlatform only)")
+  }
+
+  import autoImport._
+
+  override def globalSettings: Seq[Setting[?]] = Seq(
+    tlBspPlatforms := Set(JVMPlatform)
+  )
+
+  override def projectSettings: Seq[Setting[?]] = Seq(
+    // Keep `bspEnabled := true` for non-sbtcrossproject projects.
+    bspEnabled := crossProjectPlatform.?.value.forall(tlBspPlatforms.value.contains)
```

Comment:
This is good, but to confirm my understanding this plugin will only be enabled for `CrossProject`s anyway right? Because of the `requires: Plugins = CrossPlugin`.

Reply:
Yes, that's correct (sorry for long reply). So it should be pretty much safe simply to reference it via
```
crossProjectPlatform.value
```
I'm just not sure if it is possible to have `CrossPlugin` required but `crossProjectPlatform`.
Strictly saying, it is all up to the `CrossPlugin` implementation.
So I think maybe it makes sense to handle such a possibility, just in case.